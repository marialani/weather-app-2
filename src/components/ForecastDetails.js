import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity } = forecast;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__maxTemperature">
        {`Max temp: ${temperature.max}`}&deg;C
      </div>
      <div className="forecast-details__minTemperature">
        {`Min temp: ${temperature.min}`}&deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__windSpeed">
        Wind speed: {wind.speed}mph
      </div>
      <div className="forecast-details__windDirection">
        Wind direction: {wind.direction}
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
