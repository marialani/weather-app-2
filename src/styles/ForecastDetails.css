.forecast-details {
    display: block;
    text-align: center;
    padding: 2rem;
    letter-spacing: 1px;
    font-weight: bold;

}

.forecast-details__date {
    font-size: 25px;
}

.forecast-details__maxTemperature {
    padding: 0.5rem;

}

.forecast-details__minTemperature {
    padding: 0.5rem;

}

.forecast-details__humidity {
    padding: 0.5rem;


}

.forecast-details__windSpeed {
    padding: 0.5rem;
}

.forecast-details__windDirection {
    padding: 0.5rem;
}