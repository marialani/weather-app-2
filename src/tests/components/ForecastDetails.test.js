import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1525046400000,
    temperature: {
      min: 3,
      max: 14,
    },
    humidity: 12,
    wind: {
      speed: 20,
      direction: "n",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct value for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Mon 30th Apr")).toHaveClass("forecast-details__date");
    expect(getByText("Min temp:3°C")).toHaveClass(
      "forecast-details__minTemperature"
    );
    expect(getByText("Max temp:14°C")).toHaveClass(
      "forecast-details__maxTemperature"
    );
    expect(getByText("Humidity: 12%")).toHaveClass(
      "forecast-details__humidity"
    );
    expect(getByText("Wind speed: 20mph")).toHaveClass(
      "forecast-details__windSpeed"
    );
    expect(getByText("Wind direction: n")).toHaveClass(
      "forecast-details__windDirection"
    );
  });
});
