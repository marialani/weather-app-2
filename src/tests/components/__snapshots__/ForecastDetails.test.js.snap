// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon 30th Apr
    </div>
    <div
      class="forecast-details__maxTemperature"
    >
      Max temp:14°C
    </div>
    <div
      class="forecast-details__minTemperature"
    >
      Min temp:3°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 12%
    </div>
    <div
      class="forecast-details__windSpeed"
    >
      Wind speed: 20mph
    </div>
    <div
      class="forecast-details__windDirection"
    >
      Wind direction: n
    </div>
  </div>
</DocumentFragment>
`;
